modules/orchestrator/endpoints/addon_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package endpoints
+
+import (
+	"math/rand"
+	"reflect"
+	"sync"
+	"testing"
+
+	"bou.ke/monkey"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle"
+	"github.com/erda-project/erda/modules/orchestrator/dbclient"
+	"github.com/erda-project/erda/modules/orchestrator/services/addon"
+)
+
+func TestConcurrentWriteAddonInfos(t *testing.T) {
+	keys := []string{"a", "b", "c", "d", "e"}
+
+	var bdl *bundle.Bundle
+	monkey.PatchInstanceMethod(reflect.TypeOf(bdl), "QueryExtensions",
+		func(*bundle.Bundle, apistructs.ExtensionQueryRequest) ([]apistructs.Extension, error) {
+			var ext []apistructs.Extension
+			for i, v := range keys {
+				ext = append(ext, apistructs.Extension{ID: uint64(i), Name: v})
+			}
+			return ext, nil
+		},
+	)
+	var (
+		wg sync.WaitGroup
+		n  = 2000
```

Comment:
please reduce it，unit test no need for so much gorouting

Reply:
now change to 20